consensus/istanbul/backend/pos.go

```diff
@@ -133,43 +141,47 @@ func (sb *Backend) updateValidatorScores(header *types.Header, state *state.Stat
 	return uptimes, nil
 }
 
-func (sb *Backend) distributeEpochPayments(header *types.Header, state *state.StateDB, valSet []istanbul.Validator, maxPayment *big.Int) (*big.Int, error) {
-	totalEpochPayments := big.NewInt(0)
+func (sb *Backend) distributeValidatorRewards(header *types.Header, state *state.StateDB, valSet []istanbul.Validator, maxReward *big.Int) (*big.Int, error) {
+	totalValidatorRewards := big.NewInt(0)
 	for _, val := range valSet {
-		sb.logger.Debug("Distributing epoch payment for address", "address", val.Address())
-		epochPayment, err := validators.DistributeEpochPayment(header, state, val.Address(), maxPayment)
+		sb.logger.Debug("Distributing epoch reward for validator", "address", val.Address())
+		validatorReward, err := validators.DistributeEpochReward(header, state, val.Address(), maxReward)
 		if err != nil {
-			return totalEpochPayments, nil
+			return totalValidatorRewards, nil
 		}
-		totalEpochPayments.Add(totalEpochPayments, epochPayment)
+		totalValidatorRewards.Add(totalValidatorRewards, validatorReward)
 	}
-	return totalEpochPayments, nil
+	return totalValidatorRewards, nil
 }
 
-func (sb *Backend) distributeEpochRewards(header *types.Header, state *state.StateDB, valSet []istanbul.Validator, maxTotalRewards *big.Int, uptimes []*big.Int) (*big.Int, error) {
-	totalEpochRewards := big.NewInt(0)
-
-	// Fixed epoch reward to the community fund.
-	// TODO(asa): This should be a fraction of the overall reward to stakers.
-	communityEpochReward := big.NewInt(params.Ether)
+func (sb *Backend) distributeCommunityRewards(header *types.Header, state *state.StateDB, communityReward *big.Int) (*big.Int, error) {
+	totalCommunityRewards := big.NewInt(0)
+	// Fixed epoch reward to the infrastructure fund.
+	// TODO(joshua): The community reward needs to be split among different recipients
```

Comment:
Which recipients?

Reply:
If the reserve is low, all to the reserve.
If the reserve if fine: 2% to the eco fund & 98% to the community fund is planned. I might send it all to the governance contract right now.